astropy/cosmology/core.py

```diff
@@ -266,6 +267,12 @@ def __init_subclass__(cls):
         parameters = ordered + parameters  # place "unordered" at the end
         cls.__parameters__ = tuple(parameters)
 
+        # -------------------
+        # Register I/O
+        from astropy.cosmology.io.yaml import register_yaml
+
+        register_yaml(cls)
```

Comment:
Maybe this should be ``_register_yaml`` and also a bit more descriptive if it's going to be imported up out of ``cosmology.io``.

Reply:
For a future PR to public scope contents of ``cosmology/io``